include/exadg/functions_and_boundary_conditions/evaluate_functions.h

```diff
@@ -137,13 +136,22 @@ struct FunctionEvaluator<0, dim, Number>
       for(unsigned int d = 0; d < dim; ++d)
         q_point[d] = q_points[d][v];
 
-      function.set_time(time);
       value[v] = function.value(q_point);
     }
 
     return value;
   }
 
+  static inline DEAL_II_ALWAYS_INLINE //
+    dealii::Tensor<0, dim, dealii::VectorizedArray<Number>>
+    value(dealii::Function<dim> &                                     function,
+          dealii::Point<dim, dealii::VectorizedArray<Number>> const & q_points,
+          double const &                                              time)
+  {
+    function.set_time(time);
+    return value(function, q_points);
+  }
```

Comment:
Generally, I would like to see such changes in separate PRs (the change affects very general functionality of ExaDG), but I also do not want to insist on this here in this PR.

Reply:
I can do this in a separate PR. I will not have the time for that in the next few days, though.